modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPFailureTest.java

```diff
@@ -179,4 +199,260 @@ public void testPatchIdPNonExistentProperties() throws IOException {
+                .assertThat()
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body("message", equalTo("Endpoint configuration must be provided for the user defined " +
+                        "federated authenticators " + new String(Base64.getDecoder().decode(AUTHENTICATOR_ID_1)) + "."));
+    }
+
+    @Test
+    public void testAddIdPWithUserDefinedAuthenticatorWhenAuthenticatorAuthDetailsIsEmpty() throws IOException {
+
+        UserDefinedAuthenticatorPayload userDefinedAuthenticatorPayload =
+                createUserDefinedAuthenticatorPayloadWithEmptyAuthenticationProperties(AUTHENTICATOR_ID_1, ENDPOINT_URI);
+        String body = idpCreatePayload.replace(FEDERATED_AUTHENTICATOR_ID_PLACEHOLDER,
+                userDefinedAuthenticatorPayload.getAuthenticatorId());
+        body = body.replace(FEDERATED_AUTHENTICATOR_PLACEHOLDER,
+                userDefinedAuthenticatorPayload.convertToJasonPayload());
+
+        Response response = getResponseOfPostNoFilter(IDP_API_BASE_PATH, body);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body("message", equalTo("Invalid Request"))
+                .body("description", equalTo("Property authentication cannot be null."));
+    }
+
+    @Test
+    public void testAddIdPWithUserDefinedAuthenticatorWithExistingAuthenticatorName() throws IOException {
+
+        userDefinedAuthenticatorPayload = createUserDefinedAuthenticatorPayload(AUTHENTICATOR_ID_1, ENDPOINT_URI,
+                "testUser", "testPassword");
+        Response response = createUserDefAuthenticator("CustomAuthIDP1", userDefinedAuthenticatorPayload);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_CREATED)
+                .header(HttpHeaders.LOCATION, notNullValue());
+        String location = response.getHeader(HttpHeaders.LOCATION);
+        assertNotNull(location);
+        String customIdPId = location.substring(location.lastIndexOf("/") + 1);
+        assertNotNull(customIdPId);
+
+        // duplicate the authenticator creation
+        duplicatedUserDefinedAuthenticatorPayload =
+                createUserDefinedAuthenticatorPayload(AUTHENTICATOR_ID_1,
+                        "https://xyz.com/authenticate",
+                        "testUser1",
+                        "testPassword1");
+        Response responseOfDuplicate = createUserDefAuthenticator("CustomAuthIDP2",
+                duplicatedUserDefinedAuthenticatorPayload);
+        responseOfDuplicate.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body("message", equalTo("Federated authenticator name " +
+                        new String(Base64.getDecoder().decode(AUTHENTICATOR_ID_1)) + " is already taken."));
```

Comment:
Delete the created idp in this method

Reply:
This is updated with 104bc9c032e465bb70bd9f8c1f0dd364ba62aa37